src/pages/ProfilesExplorerView/infrastructure/labels/LabelsDataSource.ts

```diff
@@ -9,37 +9,30 @@ import {
 } from '@grafana/data';
 import { RuntimeDataSource, sceneGraph } from '@grafana/scenes';
 import { isPrivateLabel } from '@shared/components/QueryBuilder/domain/helpers/isPrivateLabel';
-import { labelsRepository } from '@shared/infrastructure/labels/labelsRepository';
+import { LabelsRepository } from '@shared/infrastructure/labels/labelsRepository';
+import { MemoryCacheClient } from '@shared/infrastructure/MemoryCacheClient';
 
 import { GroupByVariable } from '../../domain/variables/GroupByVariable/GroupByVariable';
 import { computeRoundedTimeRange } from '../../helpers/computeRoundedTimeRange';
 import { PYROSCOPE_LABELS_DATA_SOURCE } from '../pyroscope-data-sources';
 import { DataSourceProxyClientBuilder } from '../series/http/DataSourceProxyClientBuilder';
 import { LabelsApiClient } from './http/LabelsApiClient';
 
+// we instantiate a new repository here to prevent unwanted interaction with the QueryBuilder
+// indeed, when entering the "idle" state, the QueryBuilder state machine will call cancelAllLoad()
+// that will abort any ApiClient in-flight request
+const labelsRepository = new LabelsRepository({
+  apiClient: new LabelsApiClient({ dataSourceUid: '' }),
+  cacheClient: new MemoryCacheClient(),
```

Comment:
Is it worth reusing the cache between the two instances of this?
 https://github.com/grafana/explore-profiles/blob/4ee541acbe822d92abfc9344eda4611600b1476e/src/shared/infrastructure/labels/labelsRepository.ts#L115-L118

Reply:
Indeed. By making the change, I actually realised that this fix is not the best we can do. I've just commited the new fix.